internal/provider/resource_cert_request.go

```diff
@@ -254,8 +254,13 @@ func (r *certRequestResource) Create(ctx context.Context, req resource.CreateReq
 			"ipAddresses": newState.IPAddresses,
 		})
 
-		for _, ipElem := range newState.IPAddresses.Elements() {
-			ipStr := ipElem.(types.String).ValueString()
+		ipAddresses := make([]string, 0)
```

Comment:
Nit: If we're going to instantiate the slice with a size/capacity, we should probably do it fully since we know the information upfront:

```suggestion
		ipAddresses := make([]string, 0, len(newState.IPAddresses.Elements()))
```

Although most code examples for As/ElementsAs typically instantiate via:

```go
var ipAddresses []string
```

Reply:
Good call, switched them all to nil slices 👍🏻